File: php/generate-url.php
<?php

$tractorscope_api_key = 'key_1234';

$dashboard = 'dsh_abc123';

$filters = (object) array(
	'Aggregation' => 'month',
	'Date Range' => '2020-01-01to2021-01-01',
	'User' => 1,
	'Role' => 'admin'
);

$data = rawurlencode(json_encode(
	(object) array(
		'dashboard' => $dashboard,
		'filters' => $filters
	)
));

$signature = hash_hmac('sha256', $data, $tractorscope_api_key);

$tractorscope_embed_url = 'https://app.tractorscope.com/embed/dashboard/'.$dashboard.'?data='.$data.'&signature='.$signature;

echo $tractorscope_embed_url;

?>

File: python/generate-url.py

# api key secret can be generated in account

TRACTORSCOPE_API_SECRET = 'key_1234'

import hashlib
import hmac
import json
from urllib.parse import quote

def generate_tractor_url(dashboard, filters):
    data = quote(json.dumps({'dashboard': dashboard, 'filters': filters},separators=(',', ':')))

    # Create the encrypted signature
    signature = hmac.new(
        TRACTORSCOPE_API_SECRET.encode(),
        data.encode(),
        hashlib.sha256
    ).hexdigest()

    url = (
        'https://app.tractorscope.com'
        '/embed/dashboard/' +
        dashboard +
        '?data=' +
        data +
        '&signature=' +
        signature
    )

    return url

url1 = generate_tractor_url('dsh_abc123', {'userId': 'user_123'})
url2 = generate_tractor_url('dsh_bcd234', {'spaceId': '55'})

print(url1)
print(url2)